[Camel-18635] Added ability to generate route coverage reports in HTML format

cziesman opened the pull request:
The existing version of the camel-report-maven-plugin generates route coverage statistics in a text format that, while useful, can be difficult to navigate and interpret. This PR provides the ability to generate those statistics in HTML and XLSX formats as well.

davsclaus commented:
Its a large PR - I wonder if you have considered a PR with the Excel first, and then for HTML later where we need to find a way to have a HTML page that is easy for others to maintain and don't require online access to 3rd party CSS

cziesman commented on catalog/camel-report-maven-plugin/pom.xml:
```diff
@@ -33,6 +33,15 @@
     <name>Camel :: Catalog :: Camel Report Maven Plugin</name>
     <description>Maven plugin with tools to generate reports on your Camel projects</description>
 
+    <properties>
```
Done

cziesman commented on catalog/camel-report-maven-plugin/pom.xml:
```diff
@@ -145,12 +188,82 @@
+            <version>5.9.2</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
```
Done

cziesman commented on catalog/camel-report-maven-plugin/pom.xml:
```diff
@@ -145,12 +188,82 @@
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <version>3.1.2</version>
```
Done

cziesman commented:
I removed the Excel generator code. It had a few bugs anyway and can be added back later if there is a need for it. I also removed the need for online access to CSS and JS.

davsclaus commented on catalog/camel-report-maven-plugin/pom.xml:
```diff
@@ -33,6 +33,10 @@
     <description>Maven plugin with tools to generate reports on your Camel projects</description>
 
+    <properties>
+        <poi-version>5.2.3</poi-version>
```
Can you add poi to parent pom.xml as well

cziesman replied:
Sorry, POI was related to Excel generation. I removed the dependencies and the property.

davsclaus commented:
Thanks a lot @cziesman for all this work, you are very much welcome to keep contributing and improving this report, or other areas of Camel.
